compiler/pash_runtime_daemon.py

```diff
@@ -262,10 +270,53 @@ def run(self, in_filename, out_filename):
                 self.fout = open(out_filename, "w")
 
             self.parse_and_run_cmd(input_cmd)
-            
+        
+        self.reader.close()
         shutdown()
 
 
+class UnixPipeReader:
+    def __init__(self, in_filename, blocking = True):
+        self.in_filename = in_filename
+        self.buffer = ""
+        self.blocking = blocking
+        if not self.blocking:
+            self.fin = open(self.in_filename)
+
+    def get_next_cmd(self):
+        log("Previous cmd buffer:", self.buffer)
+        log("Previous cmd buffer length:", len(self.buffer))
+        input_buffer = ""
+        if self.buffer:
+            # Don't wait on fin if cmd buffer isn't empty
+            input_buffer = self.buffer
```

Comment:
We should log that we are using the previous buffer here, instead of reading for new inputs.